controller/destination/listener.go

```diff
@@ -170,13 +171,21 @@ func (l *endpointListener) toAddrSet(addresses []*updateAddress) *pb.AddrSet {
 	return &pb.AddrSet{Addrs: addrs}
 }
 
-func (l *endpointListener) getAddrMetadata(pod *coreV1.Pod) (map[string]string, *pb.TlsIdentity) {
+func (l *endpointListener) getAddrMetadata(pod *coreV1.Pod) (map[string]string, *pb.ProtocolHint, *pb.TlsIdentity) {
 	controllerNs := pkgK8s.GetControllerNs(pod)
 	ownerKind, ownerName := l.ownerKindAndName(pod)
 	labels := pkgK8s.GetPodLabels(ownerKind, ownerName, pod)
 
+	hint := &pb.ProtocolHint {}
+
+	if vers := pkgK8s.GetControllerVersion(pod); vers != "" {
```

Comment:
I think it might be smart to check that only newer versions where we know the proxy actually would understand if we sent orig-proto requests, thought wasn't sure if that seemed like too complicated, that we currently just assume that the proxies are all always upgraded to new versions.

Reply:
I don't think there's really any harm in sending older versions a protocol hint that is disregarded, is there?

Reply:
Without checking the actual version, we could be telling newer proxies that this destination is controlled by linkerd, but it wouldn't know to downgrade h2 requests via l5d-orig-proto. Specifically, if some services are still running v18.7.1 or something.

Reply:
Ah, that's a good point.  I think we can announce this as a breaking change unless it's relatively easy to discover this now.  (This class of problem will definitely come up again).

Reply:
If we're not actually checking the version here, then I prefer to instead use the "linkerd.io/control-plane-ns" label (and existing `GetControllerNs` func) to determine whether or not the pod is meshed. That's more consistent with meshed checks throughout the rest of the codebase. And the `controllerNs` local var is already assigned in this func. I'd write this as:

```
	var hint *pb.ProtocolHint
	if controllerNs != "" {
		hint = &pb.ProtocolHint{
			Protocol: &pb.ProtocolHint_H2_{
				H2: &pb.ProtocolHint_H2{},
			},
		}
	}
```

That has the benefit of also returning `nil` if the pod is not meshed, which I think is clearer than returning an empty `ProtocolHint` struct.